elements/castbar.lua

```diff
@@ -373,37 +352,78 @@ local function CastUpdate(self, event, unit, castID, spellID)
 	end
 end
 
-local function CastStop(self, event, unit, castID, spellID)
+local function CastStop(self, event, unit, _, _, ...)
 	local element = self.Castbar
 	if(not (element.ShouldShow or ShouldShow) (element, unit)) then
 		return
 	end
 
-	if(not element:IsShown() or element.castID ~= castID or element.spellID ~= spellID) then
+	local castID, interruptedBy, empowerComplete
+	if(event == 'UNIT_SPELLCAST_STOP') then
+		castID = ...
+	elseif(event == 'UNIT_SPELLCAST_EMPOWER_STOP') then
+		empowerComplete, interruptedBy, castID = ...
+	elseif(event == 'UNIT_SPELLCAST_CHANNEL_STOP') then
+		interruptedBy, castID = ...
+	end
+
+	if(not element:IsShown() or element.castID ~= castID) then
 		return
 	end
 
+	if(element.Spark) then element.Spark:Hide() end
+
+	if(interruptedBy) then
+		if(element.Text) then element.Text:SetText(INTERRUPTED) end
+
+		element.holdTime = element.timeToHold or 0
+
+		-- force filled castbar
+		element:SetMinMaxValues(0, 1)
+		element:SetValue(1)
+	end
+
 	resetAttributes(element)
```

Comment:
Shouldn't this be done after the post-hooks? Post-hooks cannot get the `spellID` this way.

Reply:
Actually, I'll just move the resetter. Maybe someone might want it for some player-related cleanup in `Post*`s.

Reply:
> Tbf, spellID is almost always secret anyway, that's why p3lim dropped it from Post*s, it's not even used anywhere in the element either.

```lua
local function PostCastFailedOrInterrupted(castbar)
	castbar:SetStatusBarColor(0.69, 0.31, 0.31)

	local time = castbar.Time
	if time then
		time:SetText(C_Spell.GetSpellName(castbar.spellID))
	end
end
```

works for me. Thanks for the change.